common/helper.go

```diff
@@ -171,3 +171,30 @@ func SprintfList(list []string, values ...interface{}) (result []string, err err
 	}
 	return
 }
+
+type PathMatchResult struct {
+	Path   string
+	Match  bool
+	IsLink bool
+}
+
+func PathMatch(path string, patternList []string) (PathMatchResult, error) {
+	result := PathMatchResult{}
+	result.Path, _ = filepath.EvalSymlinks(path)
```

Comment:
The error returned by `filepath.EvalSymlinks` gets ignored. What happens if an error gets returned?